### PR TITLE
[in_app_purchase] Fix iOS import using quotes

### PR DESCRIPTION
**Pull request by kbot**
Fixes import not found error when compiling ios add to app module that depends on in_app_purchase

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*
https://github.com/flutter/flutter/issues/157374

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/FIAPRequestHandler.h [hunk 1]
```diff
@@ -2,9 +2,9 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#import <FLTRequestHandlerProtocol.h>
 #import <Foundation/Foundation.h>
 #import <StoreKit/StoreKit.h>
+#import "FLTRequestHandlerProtocol.h"
```
Comment (jmagman) [inline packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/FIAPRequestHandler.h:7]: @LouiseHsu All of these headers are public due to this (I know this predates your migration):
https://github.com/flutter/packages/blob/a87eddf0c281d5d017093125c1a8d9161257741f/packages/in_app_purchase/in_app_purchase_storekit/darwin/in_app_purchase_storekit.podspec#L21

https://guides.cocoapods.org/syntax/podspec.html#public_header_files

Public headers should not use file (quote) imports, but should use angle brackets.
https://developer.apple.com/documentation/xcode/build-settings-reference#Quoted-Include-In-Framework-Header

Also the [import order](https://google.github.io/styleguide/objcguide.html#order-of-includes) should be
```
#import <Foundation/Foundation.h>
#import <StoreKit/StoreKit.h>
```
then
```suggestion
#import <in_app_purchase_storekit.framework/FLTRequestHandlerProtocol.h>
```

When in doubt, open up an Apple header and copy their pattern.  For example, this is StoreKit's `SKAdImpression` header:
```objc
//
//  SKAdImpression.h
//  StoreKit
//
//  Copyright © 2020 Apple Inc. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <StoreKit/StoreKitDefines.h>
```

1. Can you think through [which headers should actually be public](https://google.github.io/styleguide/objcguide.html#keep-the-public-api-simple), and only expose that from the podspec (and later from the Package.swift file)?
1. Audit the headers to make them angle bracket imports.
2. If it's done in this PR, make sure the CHANGELOG and pubspec version is also update.

Comment (LouiseHsu) [reply packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/FIAPRequestHandler.h:7]: Ah. I missed this comment. Ill make an issue to fix all the public headers.

**Comment by jmagman** [comment]
@kbot the [checks are failing ](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8733341812989654849/+/u/Run_package_tests/CHANGELOG_and_version_validation/stdout)because you need to also update the version and CHANGELOG.  

See https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version-and-changelog-updates

**Comment by stuartmorgan** [comment]
This is on `main`, which means we won't be able to update it ourselves. @kbot Are you planning on updating this to address the comments above?

**Comment by LouiseHsu** [comment]
Thank you for your PR! I'm closing this for now since there hasn't been a response, but the points raised by this PR and the subsequent comments are captured here https://github.com/flutter/flutter/issues/159291

**Pull request closed without merging**